linode/manager
# Cancellable Requests w/ Pagey

### Purpose

Create cancellable requests with paginated entities that leverage the `<Pagey />` HOC.

### Notes
* To accomplish this, `<Pagey />` will now take a function that returns the following

```
{
   request: () => promise,
   cancel: (() => void) | null,
}
```

Passing `null` as the value for the cancel property means this request is not cancellable, and `<Pagey />` will not try to cancel the request on `componentDidUnmount`

* The Axios Interceptor is adding a `isCancelled` property to the `catch` data if the cancel token exists. `<Pagey />` will check for it's existence before setting state
   * This also means that if you're cancelling a request without the use of `<Pagey />`, you need to do the following in your request on `componentDidMount`

```
 componentDidMount() {
   asyncRequest()
       .then()
       .catch(e => {
             if(e.isCancelled) { return; }
             this.setState({message: 'aaahhhhhhhhhhh'});
        })
 }
```

### To Test
1. Open up Chrome dev tools then go to the _Performance_ tab
2. Change the _Network_ setting from _Online_ to _Slow 3G_
3. Switch between SSH Keys and Referrals tab.
4. Observe the network requests in the console. You should see them cancelling
5. You should also see no `no-op setState` warnings
